spring-cloud-gcp-bigquery/src/test/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java

```diff
@@ -85,6 +89,46 @@ public void setup() {
 		this.bigquery.delete(TableId.of(DATASET_NAME, TABLE_NAME));
 	}
 
+	@Test
+	public void testLoadFileWithSchema() throws InterruptedException, ExecutionException {
+		Schema schema = Schema.of(
+				Field.newBuilder("CountyId", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+				Field.newBuilder("State", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+				Field.newBuilder("County", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build()
+		);
+
+		HashMap<String, Object> messageHeaders = new HashMap<>();
+		messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, TABLE_NAME);
+		messageHeaders.put(BigQuerySpringMessageHeaders.FORMAT_OPTIONS, FormatOptions.csv());
+		messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_SCHEMA, schema);
+
+		Message<File> message = MessageBuilder.createMessage(
+				new File("src/test/resources/data.csv"),
+				new MessageHeaders(messageHeaders));
+
+		ListenableFuture<Job> jobFuture =
+				(ListenableFuture<Job>) this.messageHandler.handleRequestMessage(message);
+
+		// Assert that a BigQuery polling task is scheduled successfully.
+		await().atMost(Duration.FIVE_SECONDS)
+				.untilAsserted(
+						() -> assertThat(
+								this.taskScheduler.getScheduledThreadPoolExecutor().getQueue()).hasSize(1));
```

Comment:
Is it possible that concurrent tests would mean this executor pool has size > 1? Is there something more meaningfully checked in the queue item?

Reply:
Hmm as long as the tests in this file are not concurrent it should be okay. It should not interfere with any other tests run in other files since the pool is not shared.

Yeah the purpose of the check is just to verify a new `Runnable` was created in the task scheduler. Doesn't seem to be anything deeper I can verify about the object though.